pinot-core/src/main/java/org/apache/pinot/core/operator/query/SelectionOperator.java

```diff
@@ -42,212 +39,168 @@
-        List<SelectionSort> sortSequence = _selection.getSelectionSortSequence();
-        int numSortColumns = sortSequence.size();
-        for (int i = 0; i < numSortColumns; i++) {
-          int ret = 0;
-          SelectionSort selectionSort = sortSequence.get(i);
-          int index = _orderByIndices.get(i);
-          // Only compare single-value columns.
-          if (!_expressionResultMetadata[index].isSingleValue()) {
-            continue;
-          }
-
-          Serializable v1 = o1[index];
-          Serializable v2 = o2[index];
-
-          DataType dataType = _expressionResultMetadata[index].getDataType();
-          switch (dataType) {
-            case INT:
-              if (!selectionSort.isIsAsc()) {
-                ret = ((Integer) v1).compareTo((Integer) v2);
-              } else {
-                ret = ((Integer) v2).compareTo((Integer) v1);
-              }
-              break;
-            case LONG:
-              if (!selectionSort.isIsAsc()) {
-                ret = ((Long) v1).compareTo((Long) v2);
-              } else {
-                ret = ((Long) v2).compareTo((Long) v1);
-              }
-              break;
-            case FLOAT:
-              if (!selectionSort.isIsAsc()) {
-                ret = ((Float) v1).compareTo((Float) v2);
-              } else {
-                ret = ((Float) v2).compareTo((Float) v1);
-              }
-              break;
-            case DOUBLE:
-              if (!selectionSort.isIsAsc()) {
-                ret = ((Double) v1).compareTo((Double) v2);
-              } else {
-                ret = ((Double) v2).compareTo((Double) v1);
-              }
-              break;
-            case BOOLEAN:
-            case STRING:
-              if (!selectionSort.isIsAsc()) {
-                ret = ((String) v1).compareTo((String) v2);
-              } else {
-                ret = ((String) v2).compareTo((String) v1);
-              }
-              break;
-            case BYTES:
-              if (!selectionSort.isIsAsc()) {
-                ret = ByteArray.compare((byte[]) v1, (byte[]) v2);
-              } else {
-                ret = ByteArray.compare((byte[]) v2, (byte[]) v1);
-              }
-            default:
-              break;
-          }
+  // TODO: Optimize the comparator by comparing dictionary ids
+  private Comparator<Serializable[]> getComparator() {
+    return (o1, o2) -> {
+      int numSortExpressions = _sortSequence.size();
+      for (int i = 0; i < numSortExpressions; i++) {
+        // Only compare single-value columns
+        if (!_blockValSets[i].isSingleValue()) {
+          continue;
+        }
 
-          if (ret != 0) {
-            return ret;
-          }
+        Serializable v1 = o1[i];
+        Serializable v2 = o2[i];
+
+        int ret = 0;
+        SelectionSort selectionSort = _sortSequence.get(i);
+        switch (_blockValSets[i].getValueType()) {
+          case INT:
+            if (!selectionSort.isIsAsc()) {
```

Comment:
Nit: can it not be replaced by one-liner:
`ret = (!selectionSort.isAsc()) ? ... : ...`